kasa/interfaces/lightpreset.py

```diff
@@ -0,0 +1,89 @@
+"""Module for LightPreset base class."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Sequence
+
+from ..feature import Feature
+from ..module import Module
+from .light import LightState
+
+
+class LightPresetInfo(ABC):
+    """Light configuration preset."""
+
+    brightness: int
+
+    hue: int | None
+    saturation: int | None
+    color_temp: int | None
+
+
+class LightPreset(Module):
+    """Base interface for light preset module."""
+
+    PRESET_NOT_SET = "Not set"
+
+    def _initialize_features(self):
+        """Initialize features."""
+        device = self._device
+        self._add_feature(
+            Feature(
+                device,
+                id="light_preset",
+                name="Light preset",
+                container=self,
+                attribute_getter="preset",
+                attribute_setter="set_preset",
+                category=Feature.Category.Config,
+                type=Feature.Type.Choice,
+                choices_getter="preset_list",
+            )
+        )
+
+    @property
+    @abstractmethod
+    def preset_list(self) -> list[str]:
+        """Return built-in effects list.
+
+        Example:
+            ['Off', 'Preset 1', 'Preset 2', ...]
+        """
+
+    @property
+    @abstractmethod
+    def preset_states_list(self) -> Sequence[LightState]:
+        """Return built-in effects list.
+
+        Example:
+            ['Off', 'Preset 1', 'Preset 2', ...]
+        """
+
+    @property
+    @abstractmethod
+    def preset(self) -> str:
+        """Return current preset name."""
+
+    @abstractmethod
+    async def set_preset(
+        self,
+        preset_name: str,
+    ) -> None:
+        """Set a light preset for the device.
+
+        The device doesn't store an active effect while not enabled so store locally.
```

Comment:
This reads a bit odd. Perhaps the name should be something like `activate_preset`or `activate_preset_by_name`?

Reply:
`The device doesn't store an active effect while not enabled so store locally.` This is a left over comment from the light effect module and will be removed.

As this is setter method for the `preset` attribute of the choice feature I think this is probably an appropriate name.  So we have `preset`, `set_preset` and `preset_list`. It is also consistent with the `LightEffect` modules which has `effect`, `set_effect` and `effect_list`.

Reply:
Okay, sounds good to me!